packages/plugin-ext/src/plugin/plugin-manager.ts

```diff
@@ -0,0 +1,131 @@
+/********************************************************************************
+ * Copyright (C) 2018 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { PluginManagerExt, PluginInitData, PluginManager, Plugin, PluginAPI } from '../api/plugin-api';
+import { PluginMetadata } from '../common/plugin-protocol';
+import * as theia from '@theia/plugin';
+
+import { dispose } from '../common/disposable-util';
+
+export interface PluginHost {
+
+    // tslint:disable-next-line:no-any
+    loadPlugin(contextPath: string, plugin: Plugin): any;
+
+    init(data: PluginMetadata[]): [Plugin[], Plugin[]];
+}
+
+interface StopFun {
```

Comment:
If you don't mind let it be `StopFn` or `StopFunc`.

Reply:
I was thinking the same :-) let's still have some FUN :-)